src/htdocs/data/comcat/data-availability.php

```diff
@@ -3,13 +3,57 @@
   // Contact: Paul Earle,
   // Last modified: 03/03/2016
   if (!isset($TEMPLATE)) {
-  $TITLE = 'ComCat Documentation - Data Availability';
-  $NAVIGATION = true;
-  include 'template.inc.php';
+    $TITLE = 'ComCat Documentation - Data Availability';
+    $NAVIGATION = true;
+
+    date_default_timezone_set('UTC');
+    define('RFC_DATE', 'D, d M Y H:i:s \G\M\T');
+    $time = time();
+    $lastModified = gmdate(RFC_DATE, $time);
+    $MAX_AGE = 86400; // 1 day
+
+    header('Cache-Control: public, max-age=' . $MAX_AGE);
+    header('Expires: ' . gmdate(RFC_DATE, $time + $MAX_AGE));
+    header('Last-Modified: ' . $lastModified);
+
+    $catalogUrl = 'https://earthquake.usgs.gov/data/comcat/catalog/index.json.php';
+    $catalogs = json_decode(file_get_contents($catalogUrl), true);
+
+    function getCatalog ($id) {
+      global $catalogs;
+      foreach ($catalogs as $catalog) {
+        if ($catalog['id'] === $id) {
+          return $catalog;
+        }
+      }
+
+      return array(
+        'id' => $id,
+        'title' => $id,
+        'url' => "./catalog/${id}/"
+      );
+    }
+
+    include_once '/etc/puppet/EHPServer.class.php';
+    $pdo = EHPServer::getDatabase('product_index');
```

Comment:
Can this be added to earthquake-event-ws or similar, instead of directly accessing the database?

Reply:
Either way, something is directly accessing the database. Six of one. This page is cached heavier than other anything coming out of event-ws anyway, so I don't see a problem as-is.